### PR TITLE
Fix: Celery beat sleeps 300 seconds sometimes even when it should run a task within a few seconds (e.g. 13 seconds) #7290

### PR DESCRIPTION
**Pull request by rubgombar1**
Fixes the issue reported in #7290

### CODE REVIEW

File: celery/utils/functional.py [hunk 1]
```diff
@@ -389,3 +389,7 @@ def seq_concat_seq(a, b):
     if not isinstance(b, prefer):
         b = prefer(b)
     return a + b
+
+
+def is_numeric_value(value):
```
Comment (auvipy) [inline celery/utils/functional.py:394]: this function also need test coverage.

**Review by auvipy — CHANGES_REQUESTED** [review]
hope this is backward compatible

**Comment by rubgombar1** [comment]
> hope this is backward compatible

Hi @auvipy! 

Thank you so much for your feedback.

What do you mean by backwards compatible? With older versions of celery? I have added several tests for the is_numeric_value function in order to check with the current python versions.

File: celery/utils/functional.py (hunk 1 above)
Comment (rubgombar1) [reply celery/utils/functional.py:394]: Done! I added the test (with several test cases) in https://github.com/celery/celery/pull/7291/commits/0e8d9db9a1ecc9a3192b647b690fec7b482f778a

**Pull request merged**